x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -73,6 +73,10 @@ public RunAsPermission runAs() {
         return runAs;
     }
 
+    public boolean hasFieldOrDocumentLevelSecurity() {
+        return indices.hasFieldOrDocumentLevelSecurity();
+    }
```

Comment:
If the PR gets merged, this new method here can be used in `IndicesPermission#authorize` to avoid the [loop](https://github.com/elastic/elasticsearch/blob/92db6a62c0b7f827e25cb992da5bac9cfdf43f34/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java#L385) for computing DLS and FLS controls when the Roles does not have it.